### ISSUE TITLE
BUG BuildingFactory: buildings are not rendered correctly

### ISSUE THREAD

**Issue by erikvullings** [brnkhy/MapzenGo#41]
Sometimes, a building is not rendered properly. It may be that the wall normal are in the wrong direction, e.g. the wall is not drawn when looking from outside to inside, but is visible the other way round. 

For example, see the building within the red line (three times the same building, seen from the top, the bottom, and in the camera).
Lat: 51.45179
Lon: 5.481454
Range: 2
Zoom: 17
Tile Size: 100

![image](https://cloud.githubusercontent.com/assets/3140667/18915418/350b82fc-8591-11e6-9986-3fa6bdcdc130.png)

**Comment by brnkhy** [comment]
hey erik,
that's the old triangulation method failing in big/complex/concave polygons. You can also get those in water polygon in low zoom levels (big, detailed, concave etc, simply can't handle those).
I added a new and much more complex triangulation library recently (see "use triangulation.net" option under building factory script. I believe it's default for water and landuse, optional for building factory) and you won't get these error with that. 
It looks sound and complete BUT should be tested for performance under few/lot/basic/complex poly setups. I suspect that old one (iirc it was ear clipping algo) might be better for lots of basic polygons so kept both options. New one also uses extreme amount of memory but this might be my fault as I'm calling it for each building even though it probably can generate  everything at once.

tldr; check that option :)

**Comment by erikvullings** [comment]
Thanks! I will give that a try. 

As the performance for a HoloLens is an important issue, I will also take a look at the current visualisation. Maybe there is an easy fix.
